src/app/utils/deriveDatesAndCost.ts

```diff
@@ -55,3 +62,24 @@ export const deriveDuration = (startDate: Date, endDate: Date) => {
     ) + 1
   );
 };
+
+export const exchange = (
+  costCurrency: string,
+  cost: number,
+  exchangeRates: ExchangeRates
+) => {
+  switch (costCurrency) {
+    case 'ZAR':
+      return cost / exchangeRates.data['ZAR'].value;
```

Comment:
You need to put in safety checks as your keys are strings not 'ZAR' etc. explicitly set.
```ts
case 'ZAR':
  let output = cost;
  if (exchangeRates?.data?.['ZAR']) {
    output = cost / exchangeRates.data['ZAR'].value;
  }
  return output;
```